govcd/roles.go

```diff
@@ -0,0 +1,149 @@
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+type OpenApiRole struct {
+	Role   *types.Role
+	client *Client
+}
+
+// GetOpenApiRoleById retrieves role by given ID
+func (adminOrg *AdminOrg) GetOpenApiRoleById(id string) (*OpenApiRole, error) {
+	endpoint := "1.0.0/roles/"
+	minimumApiVersion, err := adminOrg.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if id == "" {
+		return nil, fmt.Errorf("empty role id")
+	}
+
+	urlRef, err := adminOrg.client.BuildOpenApiEndpoint(endpoint, id)
```

Comment:
I am just thinking to introduce some more sophisiticated version handling as @dataclouder was mentioning and lets see if it gets any better.

Reply:
Just a thought. It seems, we don't have other `Build...` functions, so maybe better to name `OpenApiBuildEndpoint`, so that all OpenAPI functions have the same prefix?

Reply:
It could be so. However @vbauzysvmware raised a wider question - do we like `OpenApi`* prefix or should we aim for something shorter? https://github.com/vmware/go-vcloud-director/pull/325#issuecomment-670015793

Reply:
`OpenApi` prefix kind'a ok, but yet name everything in this way from today is not cool. Maybe we can make package and reference by package and have still normal function names? or something else?
Just from usage perspective - when I have to write code I will have always write openApiGet... to see auto complete with good suggestions.

Reply:
> Just a thought. It seems, we don't have other `Build...` functions, so maybe better to name `OpenApiBuildEndpoint`, so that all OpenAPI functions have the same prefix?

Changed this one.

Reply:
> `OpenApi` prefix kind'a ok, but yet name everything in this way from today is not cool. Maybe we can make package and reference by package and have still normal function names? or something else?
> Just from usage perspective - when I have to write code I will have always write openApiGet... to see auto complete with good suggestions.

I doubt this is worth to split it into separate package. @lvirbalas probably got a good idea that this prefix only appears in "low level" functions. All other functions won't have it.